app/usecase/user.go

```diff
@@ -32,3 +32,25 @@ func (u User) Signup(user entities.User) (string, error) {
 
 	return id, nil
 }
+
+// Delete represents business logic
+// and will take care of deleting user.
+func (u User) Delete(user entities.UserID) error {
+	err := u.Repo.Delete(user)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Delete represents business logic
+// and will take care of deleting user.
+func (u User) IsUserExists(user entities.UserID) (bool, error) {
+	isExists, err := u.Repo.IsUserExists(user)
```

Comment:
The same point as above.
(but it's better to avoid this func at all)

Reply:
Changed the code so that existence and deletion are performed in one call to the database